presto-main/src/test/java/io/prestosql/execution/buffer/BenchmarkBlockSerde.java

```diff
@@ -0,0 +1,448 @@
+        {
+            List<SqlDecimal> values = new ArrayList<>();
+            for (int i = 0; i < ROWS; ++i) {
+                if (random.nextBoolean()) {
+                    values.add(new SqlDecimal(new BigInteger(96, random), 30, 5));
+                }
+                else {
+                    values.add(null);
+                }
+            }
+            return values.iterator();
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BigintNonRandomNoNullNullBenchmarkData
+            extends BenchmarkData
+    {
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            setup(BIGINT, createValues());
+        }
+
+        private Iterator<?> createValues()
+        {
+            List<Long> values = new ArrayList<>();
+            for (int i = 0; i < ROWS; ++i) {
+                values.add((long) i);
+            }
+            return values.iterator();
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BigintNoNullBenchmarkData
+            extends BenchmarkData
+    {
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            setup(BIGINT, createValues());
+        }
+
+        private Iterator<?> createValues()
+        {
+            List<Long> values = new ArrayList<>();
+            for (int i = 0; i < ROWS; ++i) {
+                values.add(random.nextLong());
+            }
+            return values.iterator();
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BigintWithNullBenchmarkData
+            extends BenchmarkData
+    {
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            setup(BIGINT, createValues());
+        }
+
+        private Iterator<?> createValues()
+        {
+            List<Long> values = new ArrayList<>();
+            for (int i = 0; i < ROWS; ++i) {
+                if (random.nextBoolean()) {
+                    values.add(random.nextLong());
+                }
+                else {
+                    values.add(null);
+                }
+            }
+            return values.iterator();
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BigintWithNullBenchmarkDataNonOptimized
+            extends BenchmarkData
+    {
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            setup(BIGINT, createValues());
+        }
+
+        private Iterator<?> createValues()
+        {
+            List<Long> values = new ArrayList<>();
+            for (int i = 0; i < ROWS; ++i) {
+                if (random.nextBoolean()) {
+                    values.add(random.nextLong());
+                }
+                else {
+                    values.add(null);
+                }
+            }
+            return values.iterator();
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BigintWithNullBenchmarkDataFixedWidthBlock
```

Comment:
What is the difference between this one and BigintWithNullBenchmarkDataNonOptimized?
They look the same

Reply:
That is leftover from previous experiments